docs/source/app-dev/explicit-contract-disclosure.rst

```diff
@@ -0,0 +1,150 @@
+.. Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _explicit-contract-disclosure:
+
+Explicit contract disclosure (experimental)
+###########################################
+
+Starting with Canton 2.6, explicit contract disclosure is introduced as an (experimental) feature
+that allows seamlessly delegating contract read rights to a non-stakeholder using off-ledger data distribution.
+
+.. note:: Usage of disclosed contracts in command submission by means of explicit contract disclosure
+  can be enabled by toggling the ``explicit-disclosure-unsafe`` flag in the participant configuration as exemplified below.
+  However, the feature is **experimental** and **must** not be used in production environments.
+
+::
+
+    participants {
+        participant {
+            ledger-api.explicit-disclosure-unsafe = true
+        }
+    }
+
+Contract read delegation
+------------------------
+
+Contract read delegation allows a party to acquire read rights during
+command submission over a contract of which it is neither a stakeholder nor an informee.
+
+As an example application where read delegation could be used,
+let's consider a simplified swap transaction between two parties.
+In this example, each party has a football NFT card, modeled below as ``FootballCardNft``,
```

Comment:
I believe the example would work as well with a swap of assets from two different issuers.

Reply:
My favorite example would still be disclosing reference data (e.g., Moody's ratings of securities), as there we have
1. a clear system architecture (the rater allows downloading the disclosed contract for a specific rating as a .json via an API call or its website)
2. a business case
   a. the rater provides the service of validating a rating on-demand for a small fee
   b. apps can streamline workflows that require up-to-date rating data

Reply:
> I'm not a fan of using an NFT example, and thus binding us more to the crypto hype.

I had this on the back of my mind but I was not sure how we want to position ourselves wrt to these controversial topics from the crypto space. Happy to have this invalidated and change.

Reply:
@meiersi-da @mziolekda @bame-da here is a Script modelling the new example with reference data
```
module StockExchange where

import Daml.Script
import DA.Assert
import DA.Action

template IOU
  with
    issuer: Party
    owner: Party
    value: Int
  where
    signatory issuer
    observer owner

    choice IOU_Transfer: ()
      with
        target: Party
        amount: Int
      controller owner
      do
        assert (value >= amount)
        create this with issuer = issuer, owner = target, value = amount
        if value == amount then pure ()
        else void $ create this with issuer = issuer, owner = owner, value = value - amount
        pure ()

template Asset
  with
    issuer: Party
    owner: Party
    assetName: Text
  where
    signatory issuer
    observer owner

    choice Asset_Transfer: ()
      with
        newOwner: Party
        amount: Int
      controller owner
      do
        create this with owner = newOwner
        pure ()

template PriceQuotation
  with
    issuer: Party
    assetName: Text
    value: Int
  where
    signatory issuer

    nonconsuming choice PriceQuotation_Fetch: PriceQuotation
      with fetcher: Party
      controller fetcher
      do pure this

template Offer
  with
    seller: Party
    quotationProducer: Party
    offeredAssetCid: ContractId Asset
  where
    signatory seller

    choice Offer_Accept: ()
      with
        priceQuotationCid: ContractId PriceQuotation
        buyer: Party
        buyerIou: ContractId IOU
      controller buyer
      do
        priceQuotation <- exercise
          priceQuotationCid PriceQuotation_Fetch with
            fetcher = buyer
        asset <- fetch offeredAssetCid

        -- Assert the quotation issuer and asset name
        priceQuotation.issuer === quotationProducer
        priceQuotation.assetName === asset.assetName

        _ <- exercise
          offeredAssetCid Asset_Transfer with
            newOwner = buyer
            amount = 1

        _ <- exercise
          buyerIou IOU_Transfer with target = seller, amount = priceQuotation.value
        pure ()

test_happy_flow : Script ()
test_happy_flow = script do
  bank <- allocatePartyWithHint "Bank" (PartyIdHint "Bank")
  stockExchange <- allocatePartyWithHint "StockExchange" (PartyIdHint "StockExchange")
  seller <- allocatePartyWithHint "Seller" (PartyIdHint "Seller")
  buyer <- allocatePartyWithHint "Buyer" (PartyIdHint "Buyer")

  bankId <- validateUserId "bank"
  stockExchangeId <- validateUserId "stockexchange"
  sellerId <- validateUserId "seller"
  buyerId <- validateUserId "buyer"

  createUser (User bankId (Some bank)) [CanActAs bank]
  createUser (User stockExchangeId (Some stockExchange)) [CanActAs stockExchange]
  createUser (User sellerId (Some seller)) [CanActAs seller]
  createUser (User buyerId (Some buyer)) [CanActAs buyer]

  buyerIouCid <- submit bank do
    createCmd IOU with
      issuer = bank
      owner = buyer
      value = 10

  let stockName = "company stock"

  stockCid <- submit stockExchange do
    createCmd Asset with
      issuer = stockExchange
      owner = seller
      assetName = stockName

  offerCid <- submit seller do
    createCmd Offer with
      seller = seller
      quotationProducer = stockExchange
      offeredAssetCid = stockCid

  priceQuotationCid <- submit stockExchange do
    createCmd PriceQuotation with
      issuer = stockExchange
      assetName = stockName
      value = 3

  -- submitMulti not needed (readAs empty) if the buyer adds the following contracts as disclosed in the command submission
  --   - The seller's Asset
  --   - The seller's Offer
  --   - The PriceQuotation of the stock exchange

  _ <- submitMulti [buyer] [stockExchange, seller] do
    exerciseCmd offerCid Offer_Accept with priceQuotationCid = priceQuotationCid, buyer = buyer, buyerIou = buyerIouCid

  pure ()
```

Reply:
I like it. By adding the readAs clause you have made it very clear what is going on here.

I actually would add the script but in a separate snippet with some narrative interlude. You can also factor out the setup part to a separate function so that you can add even more narrative before the submitMulti bit

I will comment on the daml code once you convert it to a PR. For instance I would make the IoU a subset of the quickstart example. Consistency matters.